BeeJet.Bot/Commands/Handlers/Steam/SteamButtonsPressedHandler.cs

```diff
@@ -5,21 +5,21 @@
 
 namespace BeeJet.Bot.Commands.Handlers.Steam
 {
-    public class SteamButtonsPressedHandler : IButtonPressedHandler
+    public class SteamButtonsPressedHandler : ButtonPressedHandler
     {
 
-        [ButtonPressedHandler("join-game-", startsWith: true)]
-        public async Task JoinGamePressed(SocketMessageComponent component)
+        [ButtonPressedHandler("join-game-id-", startsWith: true)]
+        public async Task JoinGamePressed()
         {
-            var match = new Regex("join-game-([a-zA-Z0-9\\s]*)-------([a-zA-Z0-9-\\s]*)",  RegexOptions.Compiled).Match(component.Data.CustomId);
+            var match = new Regex("join-game-id-([a-zA-Z0-9\\s]*)-------([a-zA-Z0-9-\\s]*)",  RegexOptions.Compiled).Match(Context.ComponentInteraction.Data.CustomId);
```

Comment:
Persoonlijk vind ik Regex echt altijd lastig te lezen. Ik zou, als je dit met regex wilt, misschien in een methode zetten die direct zegt wat het doet.
In dit geval valideren of de gamename correct is ?

Reply:
Discord api kent alleen een custom-id property voor een button. Dat is een string met de naam van je knop. 
Dus als je wil reageren op de druk op een knop, moet je alle logica schrijven op basis van die custom-id.
Wat die regex dus doet is de naam van de game weer uit de custom-id vissen, daar is geen andere optie voor in de discord API. Tenzij we een db hebben en een soort unique id per button gaan maken en daarbij een data object opslaan. Maar voor nu zitten we vast aan regex. 

Heb hem wel in functie afgesplitst TryParseButtonCustomId

Reply:
Zou het idd voor nu afsplitsen; wat je ook nog zou kunnen overwegen is de Regexes naar een losse 'class' afsplitsen waarbij elke regex een property wordt met een beschrijvende naam. Dat gebruik ik zelf in Ovzgen ook en dat werkt op zich redelijk :) In die regex class kun je dan evt. nog met commentaar beschrijven wat hij doet en waarom